.github/workflows/ci.yml

```diff
@@ -0,0 +1,64 @@
+name: CI
+
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+
+env:
+  NODE_VERSION: 12
+
+jobs:
+  lint:
+    name: Lint files
+    runs-on: ubuntu-latest
+    timeout-minutes: 3
+    steps:
+      - name: Check out a copy of the repo
+        uses: actions/checkout@v2
+
+      - name: Use Node.js ${{ env.NODE_VERSION }}
+        uses: actions/setup-node@v2-beta
+        with:
+          node-version: ${{ env.NODE_VERSION }}
+
+      - name: Cache npm cache
+        id: cache-dependencies
+        uses: actions/cache@v2
+        with:
+          path: ~/.npm
+          key: ${{ runner.os }}-${{ env.NODE_VERSION }}-${{ secrets.CACHE_VERSION }}-npm-cache
+          restore-keys: ${{ runner.os }}-${{ env.NODE_VERSION }}-${{ secrets.CACHE_VERSION }}-
+
+      - name: Install dependencies
+        run: npm install alex@8.0.1 markdownlint-cli@0.19.0 --global
+        if: steps.cache-dependencies.outputs.cache-hit != 'true'
+
+      - name: Lint
+        run: |
+          ./scripts/validate.sh alex
+          ./scripts/validate.sh markdownlint
+
+
+  build-app:
+    name: Build app
+    runs-on: ubuntu-latest
+    timeout-minutes: 7
+    steps:
+      - name: Check out a copy of the repo
+        uses: actions/checkout@v2
+
+      - name: Set up Ruby
+        uses: ruby/setup-ruby@v1
+        with:
+          # Gets Bundler version number from `BUNDLED WITH`
+          bundler: 'Gemfile.lock'
+          # Runs `bundle install` and caches the result automatically
+          bundler-cache: true
+
+      - name: Install gems
+        run: bundle install --jobs 4 --retry 3
```

Comment:
Sample code from `action/setup-ruby`. (Install gems in parallel, and retry up to 3 times.)